ext/msgpack/packer_class.c

```diff
@@ -68,29 +70,20 @@ VALUE MessagePack_Packer_alloc(VALUE klass)
 
 VALUE MessagePack_Packer_initialize(int argc, VALUE* argv, VALUE self)
 {
-    VALUE io = Qnil;
-    VALUE options = Qnil;
-
-    if(argc == 0 || (argc == 1 && argv[0] == Qnil)) {
-        /* Qnil */
+    VALUE io = argc >= 1 ? argv[0] : Qnil;
+    VALUE options = argc >= 2 ? argv[1] : Qnil;
 
-    } else if(argc == 1) {
-        VALUE v = argv[0];
-        if(rb_type(v) == T_HASH) {
-            options = v;
-        } else {
-            io = v;
-        }
+    if (argc > 2) {
+        rb_raise(rb_eArgError, "wrong number of arguments (%d for 0..2)", argc);
+    }
```

Comment:
IIUC, Ruby's keyword arguments (`options` here) should be the last argument, not 1st (`argv[0] for argc >= 1`) or 2nd (`argv[1] for argc >= 2`).
So, could you place `if (argc > 2)` clause at first, then check `argc >= 1` and `argc == 2`?